apisix/plugins/openwhisk.lua

```diff
@@ -104,8 +106,35 @@ function _M.access(conf, ctx)
         return 503
     end
 
+    -- parse OpenWhisk JSON response
+    -- OpenWhisk supports two types of responses, the user can return only
+    -- the response body, or set the status code and header.
+    local result, err = core.json.decode(res.body)
+
+    if err then
+        core.log.error("failed to parse openwhisk response data: ", err)
+        return 503
+    end
+
     -- setting response headers
-    core.response.set_header(res.headers)
+    if result.headers ~= nil then
+        core.response.set_header(result.headers)
+    end
+
+    -- return status code and body
+    if result.statusCode ~= nil and result.body ~= nil then
+        return result.statusCode, result.body
+    end
```

Comment:
```
if result.statusCode ~= nil then
    return result.statusCode, result.body
end
```
already covers two cases?

Reply:
changed.

Reply:
The statuscode and body may not exist simultaneously, we need to handle different cases respectively.

Reply:
Return a `nil` `result.body` is equal to `return result.statusCode`. So we can save one branch.

Reply:
Applied @shuaijinchao suggestion.